Description: Output errors on STDERR, not STDOUT
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=329970
Author: Micha Lenk <micha@lenk.info>

Index: nullmailer-1.05/src/inject.cc
===================================================================
--- nullmailer-1.05.orig/src/inject.cc	2011-03-01 21:54:40.000000000 +0000
+++ nullmailer-1.05/src/inject.cc	2011-03-01 21:54:44.000000000 +0000
@@ -71,9 +71,9 @@
   {0, 0, cli_option::flag, 0, 0, 0, 0}
 };
 
-#define fail(MSG) do{ fout << "nullmailer-inject: " << MSG << endl; return false; }while(0)
-#define fail_sys(MSG) do{ fout << "nullmailer-inject: " << MSG << ": " << strerror(errno) << endl; return false; }while(0)
-#define bad_hdr(LINE,MSG) do{ header_has_errors = true; fout << "nullmailer-inject: Invalid header line:\n  " << LINE << "\n  " MSG << endl; }while(0)
+#define fail(MSG) do{ ferr << "nullmailer-inject: " << MSG << endl; return false; }while(0)
+#define fail_sys(MSG) do{ ferr << "nullmailer-inject: " << MSG << ": " << strerror(errno) << endl; return false; }while(0)
+#define bad_hdr(LINE,MSG) do{ header_has_errors = true; ferr << "nullmailer-inject: Invalid header line:\n  " << LINE << "\n  " MSG << endl; }while(0)
 
 typedef list<mystring> slist;
 // static bool do_debug = false;
@@ -428,7 +428,7 @@
 void exec_queue()
 {
   execl(nqueue.c_str(), nqueue.c_str(), NULL);
-  fout << "nullmailer-inject: Could not exec " << nqueue << ": "
+  ferr << "nullmailer-inject: Could not exec " << nqueue << ": "
        << strerror(errno) << endl;
   exit(1);
 }
@@ -549,7 +549,7 @@
     mystring tmp(o_from);
     if(!parse_addresses(tmp, list) ||
        !parse_sender(list)) {
-      fout << "nullmailer-inject: Invalid sender address: " << o_from << endl;
+      ferr << "nullmailer-inject: Invalid sender address: " << o_from << endl;
       return false;
     }
   }
@@ -561,7 +561,7 @@
   bool result = true;
   for(int i = 0; i < argc; i++) {
     if(!parse_recip_arg(argv[i])) {
-      fout << "Invalid recipient: " << argv[i] << endl;
+      ferr << "Invalid recipient: " << argv[i] << endl;
       result = false;
     }
   }
@@ -576,7 +576,7 @@
      !fix_header())
     return 1;
   if(recipients.count() == 0) {
-    fout << "No recipients were listed." << endl;
+    ferr << "No recipients were listed." << endl;
     return 1;
   }
   if(!send_message())
