Description: nullmailer didn't send QUIT after a successful transfer.
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=375380
Author: Nick Leverton <nick@leverton.org>

Index: nullmailer-1.10/protocols/smtp.cc
===================================================================
--- nullmailer-1.10.orig/protocols/smtp.cc	2012-04-20 00:10:16.000000000 +0100
+++ nullmailer-1.10/protocols/smtp.cc	2012-05-12 18:03:58.000000000 +0100
@@ -53,6 +53,7 @@
   void auth_plain(void);
   void send_data(fdibuf& msg);
   void send_envelope(fdibuf& msg);
+  void send_quit();
   void send(fdibuf& msg);
 };
 
@@ -202,10 +203,16 @@
   protocol_succ(tmp.c_str());
 }
 
+void smtp::send_quit()
+{
+  docmd("QUIT", 200);
+}
+
 void smtp::send(fdibuf& msg)
 {
   send_envelope(msg);
   send_data(msg);
+  send_quit();
 }
 
 void protocol_prep(fdibuf&)
